### DISCUSSION TITLE
Newbie - Seeking to Contribute

### DISCUSSION THREAD

**Discussion by nevermore23274** [flxzt/rnote#957] [Development]
As the title suggests, I'm somewhat new to contributions in general. However, I was looking at #790 because I'm having an issue with rnote seemingly holding on to settings from previous instances. Meaning, when I change the eraser sizes in lines 52-54 in the eraserconfig.rs file, it doesn't actually update anything when I rebuild the app. (it stays at whatever I updated it to) Of note, I am using flatpak and the VSCode extension.

If it matters at all:
System - Surface Pro 8
OS - Fedora Linux 39 (Workstation Edition)
Kernel - 6.6.6-1.surface.fc39.x86_64 (used https://github.com/linux-surface/linux-surface )

So what I'm wondering is that when I rebuild the flatpak, why is it carrying over the values rather than resetting them? I was looking to try to solve #819 as I figured that may be a good starter. But unless I update the minimum (line 52) value to something higher than what I set it at, then the value stays at whatever I left it at. The default value (line 54) doesn't seem to matter as it seems there's some sort of cache being stored. (of note, when I completely delete, repull and rebuild it does start at 12 like the default value shows)

I guess the real question to dig into: What is the "correct" workflow for this? (simply used https://github.com/flxzt/rnote/blob/main/BUILDING.md but apparently am still not doing it right or am missing something)

**Comment by silvasch** [comment]
I'm not sure if I'm understanding correctly what you're asking, but I'll try to answer:

If you want to change the default value for something, it's likely that it is stored as a setting in a file to keep the value between sessions. If you're using flatpak, this file is kept in `~/.var/app/{APP_NAME}/config/glib-2.0/settings/keyfile` (for rnote when building yourself: `~/var/app/com.github.flxzt.rnote.Devel/config/glib-2.0/settings/keyfile`). If you delete this file, all values will be reset to the default value.

**Comment by Doublonmousse** [comment]
There are a few things to note

The lines you are refering to 
https://github.com/flxzt/rnote/blob/e578ec5a021d9f2a679bb08d20148faed40ca3e0/crates/rnote-engine/src/pens/pensconfig/eraserconfig.rs#L52-L54

only set the max and min value authorized for the eraser size. If you change the eraser size in rnote, you will see that you can only go as low as 1 and as high as 500. These lines seems to be the ones doing that

https://github.com/flxzt/rnote/blob/e578ec5a021d9f2a679bb08d20148faed40ca3e0/crates/rnote-ui/src/penssidebar/eraserpage.rs#L127-L129

The default value is used for a default parameter (in the `impl Default` above) but that's more of a placeholder in case there is no provided value for initialization.

https://github.com/flxzt/rnote/blob/e578ec5a021d9f2a679bb08d20148faed40ca3e0/crates/rnote-ui/src/penssidebar/eraserpage.rs#L130-L132

Here the `init` functions set the default size to the one in the const but this is immediately changed by loading settings from the settings file : 

The values for the eraser are stored in the `gschema` file and holds all settings for rnote across sessions. If you don't want to have that happen, try to see if you can prevent the `gschema` from being installed or remove it.

The values are set from these settings in these lines
https://github.com/flxzt/rnote/blob/e578ec5a021d9f2a679bb08d20148faed40ca3e0/crates/rnote-ui/src/appwindow/appsettings.rs#L349-L388

There is additional logic here to make all of the ui work to change eraser size with the 3 first line setting the default values should nothing exist yet in the gschema file.
https://github.com/flxzt/rnote/blob/e578ec5a021d9f2a679bb08d20148faed40ca3e0/crates/rnote-ui/src/strokewidthpicker/mod.rs#L93-L132

So your best bet is to find what is this `self` in `self.setter_i` (and how to access it) from the file where eraser actions are executed (there is another part of the logic that captures event, with different calls to different functions going down to this function here) :

https://github.com/flxzt/rnote/blob/e578ec5a021d9f2a679bb08d20148faed40ca3e0/crates/rnote-engine/src/pens/eraser.rs#L53

Going back a little earlier in the calls that capture the input, you'd probably be able to change the values of the eraser size as they happen. Now what's left to see is how to calculate the velocity. I know that for pen inputs, there is a smoothing algorithm that should calculate the velocity somewhere but it's probably not activated for eraser inputs, maybe that could be an hint on where to look
